### ISSUE TITLE
[BUG]Local cluster owner create Harvester cluster failed(RKE2)

### ISSUE THREAD

**Issue by DaiYuzeng** [harvester/harvester#2567]
**Describe the bug**

**To Reproduce**

1. Create Rancher standard user with `local` cluster owner
2. Go to create Harvester cluster credential
3. Go to create Harvester cluster (RKE2)
4. Select the credential just created
5. Get node resources failed, throw 500 error code
6. Typology select is empty, can not add pod affinity

**Expected behavior**

Typology filed should be inputable.

**Support bundle**

**Environment**
 - Harvester ISO version: 
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

**Comment by DaiYuzeng** [comment]
Hi! @futuretea  
I just sent a PR include an enhancement that the typology field will be inputable if the request of getting nodes is failed.

**Comment by futuretea** [comment]
panic in Harvester rancher pod：
```
2022/07/27 08:15:19 http: panic serving pipe: runtime error: invalid memory address or nil pointer dereference
goroutine 381328 [running]:
net/http.(*conn).serve.func1()
        /usr/lib64/go/1.17/src/net/http/server.go:1802 +0xb9
panic({0x37e05a0, 0x6eac120})
        /usr/lib64/go/1.17/src/runtime/panic.go:1047 +0x266
github.com/rancher/apiserver/pkg/server.(*Server).handle(0xc004097230, 0xc010720000, 0xc010720000)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20220223185512-c4e289f92e46/pkg/server/server.go:154 +0x85d
github.com/rancher/apiserver/pkg/server.(*Server).Handle(...)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20220223185512-c4e289f92e46/pkg/server/server.go:101
github.com/rancher/steve/pkg/server/handler.(*apiServer).apiHandler.func1({0x47a7958, 0xc00fc295c0}, 0x423ad01)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/rancher/steve@v0.0.0-20220218171307-a91d90251ffa/pkg/server/handler/apiserver.go:98 +0x8c
net/http.HandlerFunc.ServeHTTP(0x47dd440, {0x47a7958, 0xc00fc295c0}, 0x423add0)
        /usr/lib64/go/1.17/src/net/http/server.go:2047 +0x2f
github.com/rancher/steve/pkg/auth.ToMiddleware.func1.1({0x47a7958, 0xc00fc295c0}, 0xc00ab52400)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/rancher/steve@v0.0.0-20220218171307-a91d90251ffa/pkg/auth/filter.go:167 +0x3d4
net/http.HandlerFunc.ServeHTTP(0xc0135b6450, {0x47a7958, 0xc00fc295c0}, 0xc00cda2c10)
        /usr/lib64/go/1.17/src/net/http/server.go:2047 +0x2f
github.com/rancher/apiserver/pkg/urlbuilder.RedirectRewrite.func1({0x47b8cc8, 0xc0071cf0a0}, 0xc00ab52400)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20220223185512-c4e289f92e46/pkg/urlbuilder/redirect.go:24 +0x129
net/http.HandlerFunc.ServeHTTP(0xc00ab52300, {0x47b8cc8, 0xc0071cf0a0}, 0x423add0)
        /usr/lib64/go/1.17/src/net/http/server.go:2047 +0x2f
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0009e9440, {0x47b8cc8, 0xc0071cf0a0}, 0xc00ab52200)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
github.com/rancher/rancher/pkg/auth/requests.authedFilter.ServeHTTP({{0x4767440, 0xc003f90270}, {0x3fb4004, 0x4}, {0xc000e82e10, 0x1, 0x1}}, {0x47b8cc8, 0xc0071cf0a0}, 0xc00ab52200)
        /go/src/github.com/rancher/rancher/pkg/auth/requests/filter.go:105 +0x186
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0009e9680, {0x47b8cc8, 0xc0071cf0a0}, 0xc00ab52000)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0009e9740, {0x47b8cc8, 0xc0071cf0a0}, 0xc016779e00)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0013a0600, {0x47b8cc8, 0xc0071cf0a0}, 0xc016779c00)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0013a0b40, {0x47b8cc8, 0xc0071cf0a0}, 0xc016779a00)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
github.com/rancher/rancher/pkg/multiclustermanager.(*DeferredServer).Middleware.func1({0x47b8cc8, 0xc0071cf0a0}, 0xc0135b66c0)
        /go/src/github.com/rancher/rancher/pkg/multiclustermanager/deferred.go:110 +0xa2
net/http.HandlerFunc.ServeHTTP(0xc016779900, {0x47b8cc8, 0xc0071cf0a0}, 0x7fb130018f18)
        /usr/lib64/go/1.17/src/net/http/server.go:2047 +0x2f
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0009e95c0, {0x47b8cc8, 0xc0071cf0a0}, 0xc016779700)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0073eae40, {0x47b8cc8, 0xc0071cf0a0}, 0xc016779400)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
github.com/rancher/rancher/pkg/api/steve/aggregation.(*aggregationHandler).Middleware.func1({0x47b8cc8, 0xc0071cf0a0}, 0xc0135b65a0)
        /go/src/github.com/rancher/rancher/pkg/api/steve/aggregation/aggregation.go:52 +0x55
net/http.HandlerFunc.ServeHTTP(0xc016779300, {0x47b8cc8, 0xc0071cf0a0}, 0x2a)
        /usr/lib64/go/1.17/src/net/http/server.go:2047 +0x2f
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0009e9500, {0x47b8cc8, 0xc0071cf0a0}, 0xc016779200)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
github.com/rancher/rancher/pkg/api/steve/proxy.NewProxyMiddleware.func1.1({0x47b8cc8, 0xc0071cf0a0}, 0xc008b591d0)
        /go/src/github.com/rancher/rancher/pkg/api/steve/proxy/proxy.go:76 +0x56
net/http.HandlerFunc.ServeHTTP(0x31c234b, {0x47b8cc8, 0xc0071cf0a0}, 0x3fc57a8)
        /usr/lib64/go/1.17/src/net/http/server.go:2047 +0x2f
github.com/rancher/rancher/pkg/api/steve/proxy.RewriteLocalCluster.func1({0x47b8cc8, 0xc0071cf0a0}, 0xc016779200)
        /go/src/github.com/rancher/rancher/pkg/api/steve/proxy/proxy.go:43 +0x177
net/http.HandlerFunc.ServeHTTP(0x11, {0x47b8cc8, 0xc0071cf0a0}, 0x3fbdc65)
        /usr/lib64/go/1.17/src/net/http/server.go:2047 +0x2f
github.com/rancher/rancher/pkg/websocket.websocketHandler.ServeHTTP({{0x4770940, 0xc00409e4f8}}, {0x47b8cc8, 0xc0071cf0a0}, 0xc016779200)
        /go/src/github.com/rancher/rancher/pkg/websocket/handler.go:35 +0xa8
github.com/rancher/apiserver/pkg/middleware.ContentTypeOptions.func1({0x47b8cc8, 0xc0071cf0a0}, 0xc001ce89a0)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20220223185512-c4e289f92e46/pkg/middleware/cache.go:47 +0xf4
net/http.HandlerFunc.ServeHTTP(0x3c00020, {0x47b8cc8, 0xc0071cf0a0}, 0x11)
        /usr/lib64/go/1.17/src/net/http/server.go:2047 +0x2f
github.com/rancher/rancher/pkg/auth.SetXAPICattleAuthHeader.func1({0x47b8cc8, 0xc0071cf0a0}, 0xc016779200)
        /go/src/github.com/rancher/rancher/pkg/auth/server.go:180 +0x1ad
net/http.HandlerFunc.ServeHTTP(0x47dd398, {0x47b8cc8, 0xc0071cf0a0}, 0x423add0)
        /usr/lib64/go/1.17/src/net/http/server.go:2047 +0x2f
github.com/rancher/steve/pkg/auth.ToMiddleware.func1.1({0x47b8cc8, 0xc0071cf0a0}, 0xc016779100)
        /go/src/github.com/rancher/rancher/go/pkg/mod/github.com/rancher/steve@v0.0.0-20220218171307-a91d90251ffa/pkg/auth/filter.go:167 +0x3d4
net/http.HandlerFunc.ServeHTTP(0x0, {0x47b8cc8, 0xc0071cf0a0}, 0x46918e)
        /usr/lib64/go/1.17/src/net/http/server.go:2047 +0x2f
net/http.serverHandler.ServeHTTP({0xc0135b6420}, {0x47b8cc8, 0xc0071cf0a0}, 0xc016779100)
        /usr/lib64/go/1.17/src/net/http/server.go:2879 +0x43b
net/http.(*conn).serve(0xc0071df7c0, {0x47dd440, 0xc007f99530})
        /usr/lib64/go/1.17/src/net/http/server.go:1930 +0xb08
created by net/http.(*Server).Serve
        /usr/lib64/go/1.17/src/net/http/server.go:3034 +0x4e8
```

**Comment by TachunLin** [comment]
Verified **fixed** on Harvester `v1.0-60921953-head` with Rancher `v2.6.7-rc6`. Close this issue.

### Result
Login with cluster owner role and provision a RKE2 cluster
Now we can input the topology key in the Topology key field of the pod selector 
![image](https://user-images.githubusercontent.com/29251855/182752496-1fa49c1d-1b93-4147-9d5b-ef3a56d5bd2b.png)


### Test Information
 - Harvester ISO version: `v1.0-60921953-head`
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 1 nodes Harvester on local kvm machine
 - Rancher version: `v2.6.7-rc6`
 - Rancher dashboard-index: https://releases.rancher.com/dashboard/latest/index.html 
 - Rancher offline-preferred: Remote

### Verify Steps
1. Import Harvester from Rancher 
2. Create a standard user `local` in Rancher User & Authentication
3. Open Cluster Management page 
4. Edit cluster config 
![image](https://user-images.githubusercontent.com/29251855/182781682-5cdd3c6a-517b-4f61-980d-3ee3cab86745.png)
5. Expand Member Roles
6. Add `local` user with Cluster Owner role
![image](https://user-images.githubusercontent.com/29251855/182781823-b71ba504-6488-4581-b50d-17c333496b8c.png)
7. Create cloud credential of Harvester
8. Login with `local` user
9. Open the provisioning RKE2 cluster page 
10. Select Advanced settings
11. Add Pod Scheduling
12. Select `Pods in these namespaces`
13. Check can input Topology key value
